_episodes/06-raster-intro.md

```diff
@@ -0,0 +1,433 @@
+---
+title: "Read and visualize raster data"
+teaching: 40
+exercises: 20
+questions:
+-  "What is a raster dataset?"
+-  "How do I read and plot raster data in Python?"
+-  "How can I handle missing data?"
+
+objectives:
+-  "Describe the fundamental attributes of a raster dataset."
+-  "Explore raster attributes and metadata using Python."
+-  "Read rasters into Python using the `rioxarray` package."
+-  "Visualize single/multi-band raster data."
+
+keypoints:
+- "`rioxarray` and `xarray` are for working with multidimensional arrays like pandas is for working with tabular data."
+- "`rioxarray` stores CRS information as a CRS object that can be converted to an EPSG code or PROJ4 string."
+- "Missing raster data are filled with nodata values, which should be handled with care for statistics and visualization."
+---
+
+
+In this episode, we will introduce the fundamental principles, packages and
+metadata/raster attributes that are needed to work with raster data in Python. We will discuss some of the core metadata elements like Coordinate Reference Systems (CRS), no data values, and resolution. We will also explore missing and bad data values as stored in a raster and how Python handles these elements.
+
+We will use a Python package in this episode to work with raster data -
+[`rioxarray`](https://corteva.github.io/rioxarray/stable/),
+which is based on the popular [`rasterio`](https://rasterio.readthedocs.io/en/latest/) package for working with rasters and [`xarray`](http://xarray.pydata.org/en/stable/) for working with multi-dimensional arrays.
+
+We will also use the [`pystac`](https://github.com/stac-utils/pystac) package to load raster from the searching results from the exercise of the previous episode. This would not be necessary if you skipped it.
+
+
+> ## Introduce the Data
+>
+> In this episode, we will continue from the `search.json` file, which we saved in an exercise from the previous episode. We will use one scene in the searching results as an example, and demonstrate the data loading and visualization.
+>
+> If you have skipped the previous episode on data accessing, you can download the same raster data from the [data repository](ToDo: add link) of this lesson.
```

Comment:
```suggestion
> If you would like to work with the data for this lesson without using pystac, you can download the same raster data from the [data repository](ToDo: add link) of this lesson. This can be useful if you need to download the data ahead of time to work through the lesson if you are without internet, or if you want to work with the data in a different GIS.
```
specified why someone would want to not use pystac, removed the word "skip"